### PR TITLE
feat(`forge`): partial forge migration

### PR DESCRIPTION
**Pull request by Evalir**
Migrates forge's primitive types to alloy. Types remaining from `ethers` are `solc`, `etherscan` and `rpc` related unless highlighted for discussion.

### CODE REVIEW

File: crates/evm/src/coverage/mod.rs [hunk 1]
```diff
@@ -115,7 +115,7 @@ impl CoverageReport {
 
 /// A collection of [HitMap]s
 #[derive(Default, Clone, Debug)]
-pub struct HitMaps(pub HashMap<H256, HitMap>);
+pub struct HitMaps(pub HashMap<B256, HitMap>);
```
Comment (mattsse) [inline crates/evm/src/coverage/mod.rs:118]: why are we using B256 now?
isn't there an alias for H256? 
this can be super confusing

**Review by mattsse — APPROVED** [review]
lgtm, 
clarification on B256

File: crates/evm/src/coverage/mod.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/evm/src/coverage/mod.rs:118]: https://docs.rs/alloy-primitives/latest/alloy_primitives/aliases/type.B256.html

Comment (DaniPopes) [reply crates/evm/src/coverage/mod.rs:118]: H stands for Hash which we moved away from months ago :)

**Pull request merged**